crypto/fipsmodule/digest/digest.c

```diff
@@ -206,7 +217,13 @@ int EVP_MD_CTX_reset(EVP_MD_CTX *ctx) {
   return 1;
 }
 
-int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type, ENGINE *engine) {
+int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type, ENGINE *engine) {  
+  // Check Digest Algorithms and value of |experimental_unstable_enabled_sha3_flag|
+  if (type->type == NID_sha3_256 && *experimental_unstable_enabled_sha3_flag_bss_get() == 0) {
```

Comment:
I am not sure how to approach this since some of the functions test the ctx and other the md structs. It is good to have the tests on every function since it saves some zero initialization here. I can also remove and only do the check in the `_ex` functions.

Reply:
OK on all functions.

In terms of re-factor. What about defining:
```
int is_algorithm_allowed(int nid) {
  if (nid == NID_sha3_256 && experimental_unstable_enabled_sha3_flag_bss_get() == 1) {
    return 1;
  }

  return 0
}
```

You can call this function in some of the if-conditions.